gulpfile.js

```diff
@@ -0,0 +1,43 @@
+var gulp = require('gulp');
+var uglify = require('gulp-uglify');
+var gzip = require('gulp-gzip');
+var minifyCss = require('gulp-minify-css');
+var sourcemaps = require('gulp-sourcemaps');
+
+gulp.task('compress', function() {
+    //TODO: Serving compressed files from S3. (We're only minifying for now)
+    //To use gzip with S3 we have to intercept calls and change the url depending on the content-header
+    //Sample full compression with gzip:
+    //gulp.src('./js/**/*.js', {base: './'})
+    //.pipe(sourcemaps.init())
+    //.pipe(uglify())
+    //.pipe(sourcemaps.write())
+    //.pipe(gzip())
+    //.pipe(gulp.dest('./dist'));
+
+    /**** APP JS *****/    
+    gulp.src('./js/**/*.js', {base: './'})
+    .pipe(sourcemaps.init())
+    .pipe(uglify())
+    .pipe(sourcemaps.write())
+    .pipe(gulp.dest('./dist'));
+
+    /**** CSS *****/    
+    gulp.src('./css/**/*.css', {base: './'})
+    .pipe(sourcemaps.init())
+    .pipe(minifyCss())
+    .pipe(sourcemaps.write())
+    .pipe(gulp.dest('./dist'));
+
+    /**** VENDOR *****/
+    // js
+    gulp.src('./vendor/**/*.js', {base: './'})
+    .pipe(uglify())
```

Comment:
Is there a reason sourcemaps aren't being used here? Seems like it'd be useful, especially since p5 in particular is a relatively young library and it'd be great to be able to see where it might be throwing errors.